syntax="proto3";

service NumStore {
    rpc SetNum(SetNumRequest) returns (SetNumResponse) {}
    rpc Fact(FactRequest) returns (FactResponse) {}
}

message SetNumRequest {
    string key=1;
    int64 value=2;
}

message SetNumResponse {
    int64 total=1;
}

message FactRequest {
    string key=1;
}

message FactResponse {
    int64 value=1;
    bool hit=2;
    string error=3;
}